### ISSUE TITLE
@UIScoped UI together with @PreserveOnRefresh behaves differently with standard Vaadin

### ISSUE THREAD

**Issue by peterl1084** [peholmst/vaadin4spring#59]
In standard Vaadin 7.2 if UI is annotated with @PreserveOnRefresh the init() method is invoked once for every UI instance that is instantiated for each diferent browser tab. This is because each tab has different UI instance and every one of them gets initialized only once as the UI's are intended to be preserved (with the annotation). Subsequent refresh on the browser tab will cause invocation to refresh() method of the same already existing UI instance.

With Vaadin4Spring the behavior is different when UI is @UIScoped and @PreserveOnRefresh. With this combination SAME UI instance is returned for each browser tab regardless of the requirement that UI instance per tab still exists. If new tab is opened same UI instance is acquired and it's refresh() method is invoked. This is most likely a problem with the UIScope implementation.

What should happen is that with @UIScoped and @PreserveOnRefresh combination new UI instance should be created per browser tab and init should be called for each and every one UI once upon its creation. Subsequent refreshes should cause invocation to refresh method on correct UI instance as it behaves with standard Vaadin 7.2. What is happening is that same UI instance is handled regardless of which tab is refreshed or opened.

**Comment by LukaszByczynski** [comment]
With current upstream UI behaves correctly. I use Jetty as application server.

**Comment by peholmst** [comment]
Closing for now. If there are still problems, create a new issue.
